Ubuntu: https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/386922
Description: Remove some verbose messages printed before reading the
 configuration file. In some ways this is awkward because it makes debugging
 harder, but it's a requirement for a smooth-looking boot process; we may be
 able to do better in future.
UbuntuSpecific: Removes status messages that are arguably useful for
 debugging; we think it looks tidier, but upstream isn't interested.

diff -ur grub2-1.98-ylmf2/kern/main.c grub2-1.98-ylmf2.new/kern/main.c
--- grub2-1.98-ylmf2/kern/main.c	2010-03-07 04:51:37.000000000 +0800
+++ grub2-1.98-ylmf2.new/kern/main.c	2011-05-25 16:34:54.176204964 +0800
@@ -154,11 +154,6 @@
   /* First of all, initialize the machine.  */
   grub_machine_init ();
 
-  /* Hello.  */
-  grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
-  grub_printf ("Welcome to GRUB!\n\n");
-  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);
-
   /* Load pre-loaded modules and free the space.  */
   grub_register_exported_symbols ();
 #ifdef GRUB_LINKER_HAVE_INIT
diff -ur grub2-1.98-ylmf2/kern/rescue_reader.c grub2-1.98-ylmf2.new/kern/rescue_reader.c
--- grub2-1.98-ylmf2/kern/rescue_reader.c	2010-03-07 04:51:37.000000000 +0800
+++ grub2-1.98-ylmf2.new/kern/rescue_reader.c	2011-05-25 16:35:18.586204641 +0800
@@ -72,8 +72,6 @@
 void
 grub_rescue_run (void)
 {
-  grub_printf ("Entering rescue mode...\n");
-
   while (1)
     {
       char *line;
diff -ur grub2-1.98-ylmf2/normal/menu.c grub2-1.98-ylmf2.new/normal/menu.c
--- grub2-1.98-ylmf2/normal/menu.c	2011-05-25 16:34:08.819538899 +0800
+++ grub2-1.98-ylmf2.new/normal/menu.c	2011-05-25 16:36:44.529536835 +0800
@@ -489,12 +489,9 @@
 
 /* Callback invoked immediately before a menu entry is executed.  */
 static void
-notify_booting (grub_menu_entry_t entry,
+notify_booting (grub_menu_entry_t entry __attribute__((unused)),
 		void *userdata __attribute__((unused)))
 {
-  grub_printf ("  ");
-  grub_printf_ (N_("Booting \'%s\'"), entry->title);
-  grub_printf ("\n\n");
 }
 
 /* Callback invoked when a default menu entry executed because of a timeout



